### PR TITLE
Enforce flow header via eslint

### LINKED ISSUE #3680: Add linting rule to enforce flow header
From time to time, new files are missing flow headers which mostly goes by unnoticed. For test files, we can still add an eslint-disable comment if we don't want flow in there.

### PR DESCRIPTION
**Pull request by philippotto**
Should be good to go, since I only added comments.

### Issues:
- fixes #3680

- [X] Ready for review

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -74,6 +74,7 @@
     "react/prop-types": "off",
     "react/require-default-props": "off",
     "import/newline-after-import": "off",
+    "flow-header/flow-header": 2,
```
Comment (daniel-wer) [inline .eslintrc.json]: Did you find out what the 2 means?

File: tools/proxy/proxy.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// @noflow
+/* eslint-disable */
```
Comment (daniel-wer) [inline tools/proxy/proxy.js]: Is this needed now?

File: .eslintrc.json (hunk 1 above)
Comment (philippotto) [reply .eslintrc.json]: Yeah, it means "error". I replaced it by "error", since this more clear.

File: tools/proxy/proxy.js (hunk 2 above)
Comment (philippotto) [reply tools/proxy/proxy.js]: Yes, both is needed. Apparently, eslint-disable doesn't act "early enough" for the flow-lint rule. Also, the file contains other eslint issues.

**Pull request merged**